tasmota/tasmota_xdrv_driver/xdrv_92_esp32_gnss.ino

```diff
@@ -0,0 +1,876 @@
+  {
+    return len;
+  }
+
+  static void handleDisconnect(AsyncClient *client)
+  {
+    AddLog(LOG_LEVEL_INFO, PSTR("GNSS: Client disconnected"));
+    for (auto &cli : streamClients)
+    {
+      if (cli.client == client)
+      {
+        cli.isActive = false;
+        break;
+      }
+    }
+  }
+};
+#endif // USE_AUX_WEBSERVER
+
+/*********************************************************************/
+/*                  SERIAL CONFIG STRING MAPPING                     */
+/*********************************************************************/
+const SerialConfigMap kSerialConfigs[] = {
+    {PSTR("SERIAL_5N1"), SERIAL_5N1},
+    {PSTR("SERIAL_6N1"), SERIAL_6N1},
+    {PSTR("SERIAL_7N1"), SERIAL_7N1},
+    {PSTR("SERIAL_8N1"), SERIAL_8N1},
+    {PSTR("SERIAL_5N2"), SERIAL_5N2},
+    {PSTR("SERIAL_6N2"), SERIAL_6N2},
+    {PSTR("SERIAL_7N2"), SERIAL_7N2},
+    {PSTR("SERIAL_8N2"), SERIAL_8N2},
+    {PSTR("SERIAL_5E1"), SERIAL_5E1},
+    {PSTR("SERIAL_6E1"), SERIAL_6E1},
+    {PSTR("SERIAL_7E1"), SERIAL_7E1},
+    {PSTR("SERIAL_8E1"), SERIAL_8E1},
+    {PSTR("SERIAL_5E2"), SERIAL_5E2},
+    {PSTR("SERIAL_6E2"), SERIAL_6E2},
+    {PSTR("SERIAL_7E2"), SERIAL_7E2},
+    {PSTR("SERIAL_8E2"), SERIAL_8E2},
+    {PSTR("SERIAL_5O1"), SERIAL_5O1},
+    {PSTR("SERIAL_6O1"), SERIAL_6O1},
+    {PSTR("SERIAL_7O1"), SERIAL_7O1},
+    {PSTR("SERIAL_8O1"), SERIAL_8O1},
+    {PSTR("SERIAL_5O2"), SERIAL_5O2},
+    {PSTR("SERIAL_6O2"), SERIAL_6O2},
+    {PSTR("SERIAL_7O2"), SERIAL_7O2},
+    {PSTR("SERIAL_8O2"), SERIAL_8O2}};
+
+/*********************************************************************/
+/*                         HELPER FUNCTIONS                          */
+/*********************************************************************/
+static const char *SerialConfigToString(uint32_t config)
+{
+  for (const auto &mapping : kSerialConfigs)
+  {
+    if (mapping.config == config)
+    {
+      return mapping.name;
+    }
+  }
+  return "UNKNOWN";
+}
+
+static uint32_t StringToSerialConfig(const char *name)
+{
+  for (const auto &mapping : kSerialConfigs)
+  {
+    if (strcasecmp(mapping.name, name) == 0)
+    {
+      return mapping.config;
+    }
+  }
+  return SERIAL_8N1;
+}
+
+/*********************************************************************/
+/*                   LOAD/SAVE SETTINGS FUNCTIONS                    */
+/*********************************************************************/
+void GNSSSettingsSave(void)
+{
+  uint32_t crc32 = GetCfgCrc32((uint8_t *)&GNSSSettings + 4, sizeof(GNSSSettings) - 4);
+  if (crc32 != GNSSSettings.crc32)
+  {
+    GNSSSettings.crc32 = crc32;
+
+    char filename[20];
+    snprintf_P(filename, sizeof(filename), PSTR(TASM_FILE_DRIVER), XDRV_92);
+
+    AddLog(LOG_LEVEL_DEBUG, PSTR("GNSS: Saving settings to file %s"), filename);
+
+#ifdef USE_UFILESYS
+    if (!TfsSaveFile(filename, (const uint8_t *)&GNSSSettings, sizeof(GNSSSettings)))
+    {
+      AddLog(LOG_LEVEL_INFO, D_ERROR_FILE_NOT_FOUND);
```

Comment:
In general, "AddLog" needs to add a 3 letters prefix indicating which module generated the log. Here probably "GNS: "